pkg/resources/k8sgpt.go

```diff
@@ -216,6 +216,27 @@ func GetDeployment(config v1alpha1.K8sGPT) (*appsv1.Deployment, error) {
 			},
 		},
 	}
+	if config.Spec.RemoteCache.Credentials != nil {
+		addRemoteCacheEnvVar := func(name, key string) {
+			envVar := v1.EnvVar{
+				Name: name,
+				ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: config.Spec.RemoteCache.Credentials.Name,
+						},
+						Key: key,
+					},
+				},
+			}
+			deployment.Spec.Template.Spec.Containers[0].Env = append(
+				deployment.Spec.Template.Spec.Containers[0].Env, envVar,
+			)
+		}
+		addRemoteCacheEnvVar("AWS_ACCESS_KEY_ID", config.Spec.RemoteCache.Credentials.AccessKeyID)
+		addRemoteCacheEnvVar("AWS_SECRET_ACCESS_KEY", config.Spec.RemoteCache.Credentials.SecretAccessKey)
```

Comment:
I am confused by this and by the new `CredentialsRef` type,

This should:

// Check to see if that secret exists ( missing) 
// Unwrap the secret literal values ( missing ) 
// Mount each value as an EnvVarSecret ( done )

Reply:
In my opinion, to be consistent with the rest of the code, it is necessary to check this: config.Spec.RemoteCache.Credentials != nil. As for the secret related to the OpenAI credentials, it seems to me that we are not checking the presence of the secret itself, but rather the configuration. Am I mistaken?